nrf_modem/doc/delta_dfu.rst

```diff
@@ -0,0 +1,219 @@
+
+The delta update functionality is available through the Modem library function :ref:`nrf_modem_delta_dfu_api`, found in :file:`nrfxlib/nrf_modem/include/nrf_modem_delta_dfu.h`.
+
+Delta update packages
+*********************
+
+A delta update is a binary file, with a :file:`.bin` extension.
+The update file is named in such a way that both the source and destination firmware versions can be identified, for example, :file:`mfw_nrf9160_update_from_1.2.2_to_1.2.3`.
+Modem delta update files are part of the official modem releases and can be found in the `nRF9160 modem firmware zip file`_.
+
+.. note::
+   Delta update files might not be available for any two of the modem firmware versions but only for specific versions, as applicable.
+   This might depend on the extent of the update, which cannot exceed the size of the scratch area of the source firmware version.
+
+
+Limitations
+===========
+
+The size of the scratch area can be retrieved with the :c:func:`nrf_modem_delta_dfu_area` function, which returns the size of the area in bytes.
+A limitation of delta updates is that the size of the update must not exceed the area of flash memory that a given modem firmware version has dedicated for this operation.
+The size of this area varies for any given modem firmware version and it can be retrieved with the Modem library :c:func:`nrf_modem_delta_dfu_area` function.
+In general, this function can be useful when the size of the largest delta update that can be received needs to be reported to an external service, for example an LwM2M server.
+In practice, the application does not use this function in the normal update process, since all delta update files that are part of the `nRF9160 modem firmware zip file`_ are guaranteed to fit in the scratch area of the firmware version that they apply to.
+
+Features
+********
+
+Following are the features of modem firmware delta updates.
+
+Security
+========
+
+Modem firmware delta updates implement security features to ensure that only firmware updates from `nRF9160 modem firmware zip file`_ are ever programmed to the modem.
+Delta update files are encrypted and signed.
+The modem takes care of verifying the signature and decrypting the content as necessary.
+
+Fail-safety
+===========
+
+Modem firmware delta updates implement several fail-safe mechanisms to ensure that the modem continues to operate normally in case of errors.
+
+* If the modem detects issues with the firmware that is being transferred, such as the firmware being corrupt, the modem automatically aborts the firmware update process and continues to operate as normal.
+* If the modem detects issues with the new firmware during the update process at boot time, such as the inability to communicate with the application by using the Modem library, it automatically performs a rollback to the previous firmware version and continue to operate as normal.
```

Comment:
I would omit: `such as the inability to communicate with the application by using the Modem library,` after all